### ISSUE TITLE
Dynamic provisioning tests are flaky

### ISSUE THREAD

**Issue by msau42** [kubernetes/kubernetes#49907]
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug
/sig storage

**What happened**:
Dynamic provisioning tests are flaky, perhaps due to test cleanup:

```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/volume_provisioning.go:570
Expected error:
    <*errors.errorString | 0xc4203f4d10>: {
        s: "PersistentVolume pvc-75bfa964-7611-11e7-8309-42010af0003c still exists within 20m0s",
    }
    PersistentVolume pvc-75bfa964-7611-11e7-8309-42010af0003c still exists within 20m0s
not to have occurred
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/storage/volume_provisioning.go:143
```

```
Error: 2 leaked resources
[ disks ]
+gke-bootstrap-e2e-9cfc-pvc-4f55f7b2-7613-11e7-8309-42010af0003c  us-central1-f  2        pd-standard  READY
+gke-bootstrap-e2e-9cfc-pvc-75bfa964-7611-11e7-8309-42010af0003c  us-central1-f  2        pd-standard  READY
```

[testgrid](https://k8s-testgrid.appspot.com/sig-storage#gke-slow)
[logs](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gke-slow/7817)

**What you expected to happen**:
Tests should pass

**Comment by msau42** [comment]
Seems to only flake on GKE

**Comment by msau42** [comment]
From the logs, it looks like we tried to delete the disk, but it failed because it was still in use.  But then we never tried to delete the disk again after that, even after the disk got detached.  This could be the same issue as #49905, where the pv controller resync period is too long.  Not sure why it only flakes on GKE though.

**Comment by msau42** [comment]
It's been fixed by #49016
